ddtrace/internal/rate_limiter.py

```diff
@@ -113,6 +113,7 @@ def _replenish(self, timestamp_ns):
         # type: (int) -> None
         # If we are at the max, we do not need to add any more
         if self.tokens == self.max_tokens:
+            self.last_update_ns = timestamp_ns
```

Comment:
perhaps we can move this at the function entry and remove line 122? Or do a `try...finally` and update `last_update_ns` in the finally block?

Reply:
If we move it to the function entry though our calculation for `elapsed = (timestamp_ns - self.last_update_ns) / 1e9` will be wrong though, won't it?  I'll use a `try finally` block.

Reply:
Thanks for the recommendation! I couldn't think of the best way to do it.

Reply:
> If we move it to the function entry though our calculation for `elapsed = (timestamp_ns - self.last_update_ns) / 1e9` will be wrong though, won't it? I'll use a `try finally` block.

ah right 🤦 ! I didn't look carefully enough, sorry about that!

Reply:
can you add a comment for that here? I almost made the same suggestion as Gab here.